lib/vehicle_factory.rb

```diff
@@ -0,0 +1,20 @@
+require 'date'
+
+class VehicleFactory
+  def initialize
+    @new_vehicles = []
+  end
+
+  def create_vehicles(remote_data)
+    remote_data.each do |data|
+      vehicle = Vehicle.new(
+        vin: data[:vin_1_10],
+        year: data[:model_year],
+        make: data[:make],
+        model: data[:model],
+        engine: :ev)
+      @new_vehicles << vehicle
+    end
+    @new_vehicles
+  end
+end
```

Comment:
Great job here. I see on line `18` is returning the `@new_vehicles` array as the final operation in the method. You can eliminate the need to do this if you switch out `each` for `map` like so:

```ruby
require "./lib/vehicle"

class VehicleFactory
  def create_vehicles(remote_data)
    remote_data.map do |data|
      Vehicle.new(
        vin: data[:vin_1_10],
        year: data[],
        make: data[:make],
        model: data[:model],
        engine: ev
      )
    end
  end
end
```

1. You'll see I completely got rid of the `initialize` and the `@new_vehicles` array altogether
2. Converted `each` to `map` since `map` transforms every element in an array from one thing to something else, in this case -- `Vehicle` objects
3. Don't forget to require the `Vehicle` class at the top of this file since you are instantiating new vehicles